pkg/alerts/alertsHandler/alertsHandler.go

```diff
@@ -193,34 +193,59 @@ func ProcessSilenceAlertRequest(ctx *fasthttp.RequestCtx) {
 	}
 
 	// Parse request
-	var silenceRequest struct {
-		AlertID        string `json:"alert_id"`
-		SilenceMinutes uint64 `json:"silence_minutes"`
-	}
-	if err := json.Unmarshal(ctx.PostBody(), &silenceRequest); err != nil {
+	var request alertutils.AlertSilenceRequest
+	if err := json.Unmarshal(ctx.PostBody(), &request); err != nil {
 		utils.SendError(ctx, fmt.Sprintf("Failed to unmarshal json. Error=%v", err), "", err)
 		return
 	}
 
-	// Find alert and update SilenceMinutes
-	alertDataObj, err := databaseObj.GetAlert(silenceRequest.AlertID)
+	// Find alert in the database
+	alertDataObj, err := databaseObj.GetAlert(request.AlertID)
 	if err != nil {
-		utils.SendError(ctx, fmt.Sprintf("Failed to find alert. Error=%v", err), fmt.Sprintf("alert ID: %v", silenceRequest.AlertID), err)
+		utils.SendError(ctx, fmt.Sprintf("Failed to find alert. Error=%v", err), fmt.Sprintf("alert ID: %v", request.AlertID), err)
 		return
 	}
 
-	alertDataObj.SilenceMinutes = silenceRequest.SilenceMinutes
-	// Update the SilenceMinutes
+	// Handle Silence or Unsilence based on mode
+	if mode == alertutils.SilenceMode {
+		if request.SilenceMinutes == 0 {
+			utils.SendError(ctx, "SilenceMinutes must be greater than zero", "", nil)
+			return
+		}
+		if request.SilenceEndTime == 0 {
+			utils.SendError(ctx, "SilenceEndTime must be greater than zero", "", nil)
+			return
+		}
+		alertDataObj.SilenceMinutes = request.SilenceMinutes
+		alertDataObj.SilenceEndTime = request.SilenceEndTime
+	} else if mode == alertutils.UnsilenceMode {
+		alertDataObj.SilenceMinutes = 0
+		alertDataObj.SilenceEndTime = 0
+	}
+
```

Comment:
nit: have an `else` here to log an error if we get to an unexpected state. Or perhaps change `mode` from a string to a boolean since there's only two options